seq_tools/validation/c605_all_files_accessible.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+
+"""
+    Copyright (c) 2020, Ontario Institute for Cancer Research (OICR).
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+    Authors:
+        Junjun Zhang <junjun.zhang@oicr.on.ca>
+"""
+
+
+import os
+from base_checker import BaseChecker
+
+
+class Checker(BaseChecker):
+    def __init__(self, ctx, metadata):
+        super().__init__(ctx, metadata, __name__)
+
+    def check(self):
+        files_in_subdir = self.files
+
+        if files_in_subdir is None:
+            message = "No file information available in the submission " \
+                "directory. This is likely a metadata only validation, " \
+                "should not have invoked this checker. Please ignore."
+            self.logger.info(message)
+            self.message = message
+            self.status = 'UNKNOWN'
+            return
+
+        files_missed_in_subdir = set()
+        files_unaccessbile_in_subdir = set()
+
+        files_in_metadata = self.metadata['files']
+        for f in files_in_metadata:
+            if f['fileName'] not in files_in_subdir:
```

Comment:
Can we add one more check to see if f['fileName'] exist or not? So as to avoid the failures for test cases without the field or field is Null.

Reply:
As discussed we will deal with this type of exception by introducing an `ERROR` status. At the time we perform checks on sequencing files, we will need to have the metadata to be valid enough to allow file checks to be performed, otherwise put check status to `ERROR` is reasonable.